IBM/openapi-validator
# refactor: introduce Spectral `oas3-schema` rule

Purpose:
- Simplify our ruleset by using equivalent Spectral validation.

Changes:
- Enable the `oas3-schema` rule by default
- Remove redundant in-code validation that is covered by the `oas3-schema` rule

Tests:
- Remove tests that test removed code
- Ensure validator behavior is maintained by adding tests that ensure Spectral produces equivalent error for a particular test

Docs:
- Add `oas3-schema` rule to list of Spectral rules enabled by default
